components/browser/session/src/main/java/mozilla/components/browser/session/engine/EngineObserver.kt

```diff
@@ -31,7 +32,7 @@ internal class EngineObserver(
 ) : EngineSession.Observer {
 
     override fun onLocationChange(url: String) {
-        if (session.url != url) {
+        if (!isHostEquals(session.url, url)) {
             session.title = ""
             session.icon = null
```

Comment:
I think we should only use that workaround for `icon` and not the title.

Reply:
Makes sense. I'll change that.